pagerduty/resource_pagerduty_service.go

```diff
@@ -296,12 +301,17 @@ func buildServiceStruct(d *schema.ResourceData) (*pagerduty.Service, error) {
 	if attr, ok := d.GetOk("alert_grouping_parameters"); ok {
 		service.AlertGroupingParameters = expandAlertGroupingParameters(attr)
 	} else {
-		service.AlertGroupingParameters = &pagerduty.AlertGroupingParameters{}
+		// Clear AlertGroupingParameters as it takes precedence over AlertGrouping and AlertGroupingTimeout which are apparently deprecated (that's not explicitly documented in the API)
+		service.AlertGroupingParameters = nil
 	}
-	// Using GetOkExists to allow for alert_grouping_timeout to be set to 0 if needed.
-	if attr, ok := d.GetOkExists("alert_grouping_timeout"); ok {
```

Comment:
With terraform-plugin-sdk v1.7.0 `GetOkExists` is documented as being only useful for boolean attributes:

```
// GetOkExists returns the data for a given key and whether or not the key
// has been set to a non-zero value. This is only useful for determining
// if boolean attributes have been set, if they are Optional but do not
// have a Default value.
//
// This is nearly the same function as GetOk, yet it does not check
// for the zero value of the attribute's type. This allows for attributes
// without a default, to fully check for a literal assignment, regardless
// of the zero-value for that type.
// This should only be used if absolutely required/needed.
```

https://github.com/hashicorp/terraform-plugin-sdk/blob/v1.7.0/helper/schema/resource_data.go#L111-L120

Since terraform-plugin-sdk v1.9.0, it is documented as deprecated as it leads to undefined behaviors:

```
// GetOkExists can check if TypeBool attributes that are Optional with
// no Default value have been set.
//
// Deprecated: usage is discouraged due to undefined behaviors and may be
// removed in a future version of the SDK
```

https://github.com/hashicorp/terraform-plugin-sdk/blob/v1.9.0/helper/schema/resource_data.go#L111-L115
https://github.com/hashicorp/terraform-plugin-sdk/blob/v1-maint/CHANGELOG.md#190-march-26-2020
https://github.com/hashicorp/terraform-plugin-sdk/issues/350

Reply:
Note: as this PR converts the attribute from int to string like the other similar attributes (`acknowledgement_timeout` and `auto_resolve_timeout`), the ability to set it to 0 is preserved.